src/state/opamSwitchState.ml

```diff
@@ -1305,3 +1305,169 @@ let update_repositories gt update_fun switch =
   OpamFile.Switch_config.write
     (OpamPath.Switch.switch_config gt.root switch)
     conf
+
+
+(* dependencies computation *)
+
+let dependencies_filter_to_formula ~build ~post ~test ~doc
+    ~dev_setup ~dev st nv formula =
+  let env v =
+    if List.mem v OpamPackageVar.predefined_depends_variables then
+      match OpamVariable.Full.to_string v with
+      | "dev" -> Some (B (dev && is_dev_package st nv))
+      | "with-test" -> Some (B (test))
+      | "with-doc" -> Some (B (doc))
+      | "with-dev-setup" -> Some (B (dev_setup))
+      | "build" -> Some (B build)
+      | "post" -> Some (B post)
+      | _ -> None
+    else
+    let r = OpamPackageVar.resolve_switch ~package:nv st v in
+    if r = None then
+      (if OpamFormatConfig.(!r.strict) then
+         OpamConsole.error_and_exit `File_error
+           "Undefined filter variable %s in dependencies of %s"
+       else
+         log
+           "ERR: Undefined filter variable %s in dependencies of %s")
+        (OpamVariable.Full.to_string v) (OpamPackage.to_string nv);
+    r
+  in
+  OpamFilter.filter_formula ~default:true env formula
+
+let dependencies ~depopts ~build ~post ~installed
+    ?(test=OpamStateConfig.(!r.build_test))
+    ?(doc=OpamStateConfig.(!r.build_doc))
+    ?(dev_setup=OpamStateConfig.(!r.dev_setup))
```

Comment:
Do we need all these , `universe` is already computed using them. Here, they are used to specify dependencies formula